packages/playwright-ct-react/registerSource.mjs

```diff
@@ -21,48 +21,79 @@ import * as __pwReact from 'react';
 import { createRoot as __pwCreateRoot } from 'react-dom/client';
 
 /** @typedef {import('../playwright-test/types/experimentalComponent').Component} Component */
+/** @typedef {import('../playwright-test/types/experimentalComponent').JsxComponent} JsxComponent */
+/** @typedef {import('../playwright-test/types/experimentalComponent').ObjectComponent} ObjectComponent */
 /** @typedef {import('react').FunctionComponent} FrameworkComponent */
 
+/** @type {Map<string, () => Promise<FrameworkComponent>>} */
+const __pwLoaderRegistry = new Map();
+
 /** @type {Map<string, FrameworkComponent>} */
 const __pwRegistry = new Map();
+
 /** @type {Map<Element, import('react-dom/client').Root>} */
 const __pwRootRegistry = new Map();
 
 /**
- * @param {{[key: string]: FrameworkComponent}} components
+ * @param {{[key: string]: () => Promise<FrameworkComponent>}} components
  */
 export function pwRegister(components) {
   for (const [name, value] of Object.entries(components))
-    __pwRegistry.set(name, value);
+    __pwLoaderRegistry.set(name, value);
+}
+
+/**
+ * @template Component
+ * @param {string} type
+ * @param {Map<string, Component>} componentRegistry
+ */
+function findComponent(type, componentRegistry) {
+  const Component = componentRegistry.get(type);
```

Comment:
This one seems to be looking up the registration, I.e. within the _pwLoaderRegistry. Name the method respectively and don't pass componentRegistry here, it is global. Otherwise I can get an idea that you can be passing __pwRegistry here sometimes, which would make things very confusing.